### PR TITLE
Revise test qobj headers in result sims 87

### PR DESCRIPTION
**Pull request by jwoehr**

### CODE REVIEW

**Comment by jwoehr** [comment]
Answers #87

File: test/ibmq/test_ibmq_backends.py [hunk 1]
```diff
@@ -105,11 +103,10 @@ def test_qobj_headers_in_result_sims(self, qe_token, qe_url):
             with self.subTest(backend=backend):
                 circuits = transpile(self.qc1, backend=backend)
 
-                # Update the Qobj header.
-                qobj_header = QobjHeader.from_dict(custom_qobj_header)
                 # TODO: assemble appends extra keys to the header in terra 0.8.
```
Comment (diego-plan9) [inline test/ibmq/test_ibmq_backends.py]: Can you also delete the line with the TODO?

Comment (jwoehr) [reply test/ibmq/test_ibmq_backends.py]: Done.

**Comment by diego-plan9** [comment]
Thanks @jwoehr ! It's looking good - can you also take the chance to revise `test_qobj_headers_in_result_devices`? It should probably end up containing the same lines of code (except for the `simulator=False`).

File: test/ibmq/test_ibmq_backends.py (hunk 1 above)
Comment (diego-plan9) [reply test/ibmq/test_ibmq_backends.py]: Great, thanks!

**Comment by jwoehr** [comment]
editing and testing, just a moment ...

**Comment by jwoehr** [comment]
(That should have bombed, I think it has been skipped as a "slow test")

**Comment by diego-plan9** [comment]
> (That should have bombed, I think it has been skipped as a "slow test")

Yes - just in case you have not encountered it before, locally you can enable the running of slow tests via an environment variable `QISKIT_TESTS=run_slow`. However, note that they are indeed slow (the cause of slowness is that they use the real devices, which involves having to wait in the queue for minutes in most cases) - I'd recommend using it with caution (as in, only using it for launching a particular test, not the full suite by default).

**Comment by jwoehr** [comment]
I'll try that now for good luck, since we changed a slow test.

**Comment by jwoehr** [comment]
Well, that was interesting running the slow tests. The tests I have edited ran okay, but another one failed:
```
======================================================================
FAIL: test_run_device (test.ibmq.test_ibmq_job.TestIBMQJob)
Test running in a real device.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jax/work/QISKit/DEV/qiskit-terra/qiskit/test/decorators.py", line 72, in _wrapper
    return func(*args, **kwargs)
  File "/home/jax/work/QISKit/DEV/qiskit-terra/qiskit/test/decorators.py", line 178, in _wrapper
    return decorated_func(self, *args, **kwargs)
  File "/home/jax/work/QISKit/DEV/qiskit-ibmq-provider/test/ibmq/test_ibmq_job.py", line 96, in test_run_device
    self.assertDictAlmostEqual(counts_qx, counts_ex, shots * 0.1)
AssertionError: ('10': 208 != 0), ('11': 278 != 512.0), ('01': 135 != 0), ('00': 403 != 512.0) within 102.4 delta

----------------------------------------------------------------------
```

**Pull request merged**